src/common/Evidence.h

```diff
@@ -147,7 +147,40 @@ class Evidence {
     return std::time(nullptr);
   }
 
-  // given a date time value, stripe time out to date granularity
+  // given a date time value, strip all but year
+  // such as value of (2019-05-15 23:01:34) => value of (2019-01-1 00:00:00)
+  static std::time_t year(std::time_t time) {
+    // remove all except for date
+    auto strip_days = date(time);
+    auto tm = std::gmtime(&strip_days);
+    // yday stores [0-365]
+    auto delta_days = (tm->tm_yday) * 86400; 
+    return strip_days - delta_days;
+  }
+
+  // given a date time value, strip all but month
```

Comment:
it will be great to support the quarter as well - Q1(Jan~Mar), Q2(April~Jun), Q3, Q4, a lot of financial data use these terms